lib/puppeteer/wait_task.rb

```diff
@@ -37,7 +37,7 @@ def initialize(dom_world:, predicate_body:, title:, polling:, timeout:, args: []
 
     # Since page navigation requires us to re-install the pageScript, we should track
     # timeout on our end.
-    if timeout
+    if timeout && timeout > 0
```

Comment:
Concurrent::Promises.schedule(0) works **without delay**.
We have to ensure the timeout to be set.